### DISCUSSION TITLE
Add refresh button to Genymotion Desktop GUI

### DISCUSSION THREAD

**Discussion by bhein-sherpa6** [Genymobile/feedback#61] [Genymotion Desktop] [answered]
We are currently using the Genymotion Desktop GUI in combination with gmtool to create/start/stop/destroy devices programmatically. We noticed that when a device is created via gmtool admin, it does not appear to be listed in the GUI until we close/reopen the app. Adding a refresh button (or querying for Genymotion virtual devices) would be a useful feature to resolve this.

**Comment by agavignet** [comment]
What do you think about this behavior? If this suits you, we can try to push it in the next Desktop version.

https://github.com/Genymobile/feedback/assets/16418264/0deaeb51-d127-4d46-bbc2-1b8debc980ff

Adrien

**Comment by bhein-sherpa6** [comment]
This is even better than a refresh button that needs to be clicked manually, yes. I would ensure that it also gets removed from the GUI if device is deleted through gmtool as well. Nice!

Brandon

From: Adrien Gavignet ***@***.***>
Date: Wednesday, June 28, 2023 at 10:07 AM
To: Genymobile/feedback ***@***.***>
Cc: Brandon Hein ***@***.***>, Author ***@***.***>
Subject: Re: [Genymobile/feedback] Add refresh button to Genymotion Desktop GUI (Discussion #61)
CAUTION:This email originated outside of the organization by an untrusted source.


What do you think about this behavior? If this suits you, we can try to push it in the next Desktop version.

https://github.com/Genymobile/feedback/assets/16418264/0deaeb51-d127-4d46-bbc2-1b8debc980ff<https://usg02.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.com%2FGenymobile%2Ffeedback%2Fassets%2F16418264%2F0deaeb51-d127-4d46-bbc2-1b8debc980ff&data=05%7C01%7Cbhein%40sherpa6.com%7C2b8e0d5b28084be9389d08db77f1d063%7Ca8a70f19509940c6bbe4565a74853ff4%7C1%7C0%7C638235652684563855%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=EZEBkVVH7GKbNgGabDEd4LVaTiJWwIoK%2FWphOGSn9Bw%3D&reserved=0>

**Comment by agavignet** [comment, marked as answer]
@bhein-sherpa6 The feature you've suggested is available in Desktop 3.5, hope you will enjoy it.

**Comment by bhein-sherpa6** [comment]
Yes, I’ve been using it. Great work! This is very useful and means I don’t need to restart the GUI between automation runs.

Thank you,

Brandon

From: Adrien Gavignet ***@***.***>
Date: Tuesday, August 1, 2023 at 6:25 AM
To: Genymobile/feedback ***@***.***>
Cc: Brandon Hein ***@***.***>, Mention ***@***.***>
Subject: Re: [Genymobile/feedback] Add refresh button to Genymotion Desktop GUI (Discussion #61)
CAUTION:This email originated outside of the organization by an untrusted source.


@bhein-sherpa6<https://usg02.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.com%2Fbhein-sherpa6&data=05%7C01%7Cbhein%40sherpa6.com%7Cdf9100cd94464215de3208db928a5a04%7Ca8a70f19509940c6bbe4565a74853ff4%7C1%7C0%7C638264895148120621%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=mpPIAsf5USkjRKFxTmS6Gx683pvHUB%2Fk1WeQ%2BuKAy3c%3D&reserved=0> The feature you've suggested is available in Desktop 3.5, hope you will enjoy it.
